src/Streamly/Internal/Data/Array/Stream/Mut/Foreign.hs

```diff
@@ -246,6 +246,55 @@ compactLEParserD n = ParserD.Parser step initial extract
     functionPath =
         "Streamly.Internal.Data.Array.Stream.Mut.Foreign.compactLEParserD"
 
+-- | Coalesce adjacent arrays in incoming stream to form bigger arrays of a
+-- minimum specified size. Note that if a all the arrays in the stream together
+-- are smaller than the specified size the resulting array will be smaller than
+-- the specified size. When we coalesce multiple arrays if the size would exceed
+-- the specified size we stop coalescing further.
+--
+-- /Internal/
+{-# INLINE_NORMAL compactGEFold #-}
+compactGEFold ::
+       forall m a. (MonadIO m, Storable a)
+    => Int -> FL.Fold m (Array a) (Array a)
+compactGEFold n = Fold step initial extract
+
+    where
+
+    nBytes = n * sizeOf (undefined :: a)
+
+    initial =
+        return
+            $ if n <= 0
+              then error
+                       $ functionPath
+                       ++ ": the size of arrays ["
+                       ++ show n ++ "] must be a natural number"
+              else FL.Partial Nothing
+
+    step Nothing arr =
+        return
+            $ let len = MArray.byteLength arr
+               in if len >= nBytes
+                  then FL.Done arr
+                  else FL.Partial (Just arr)
+    step (Just buf) arr = do
+        let len = MArray.byteLength buf + MArray.byteLength arr
+        buf1 <-
+            if MArray.byteCapacity buf < len
```

Comment:
Since we have collect `>= n` we can allocate at least `n` right away.

Reply:
In compactLE as well we can allocate equal to `n` to avoid repeated allocations.

Reply:
Yeah.

Reply:
In compactLE, we do it already.